utils/cli/cli.c

```diff
@@ -0,0 +1,657 @@
+				const char *path;
+				dbus_message_iter_get_basic(&iter, &path);
+				printf("PCMRemoved %s\n", path);
+				return DBUS_HANDLER_RESULT_HANDLED;
+			}
+	}
+
+	return DBUS_HANDLER_RESULT_NOT_YET_HANDLED;
+}
+
+static int cmd_monitor(int argc, char *argv[]) {
+	if (argc != 1) {
+		print_error("Invalid arguments.");
+		return EXIT_FAILURE;
+	}
+
+	bluealsa_dbus_connection_signal_match_add(&dbus_ctx,
+			service_name, NULL, BLUEALSA_INTERFACE_MANAGER, "PCMAdded", NULL);
+	bluealsa_dbus_connection_signal_match_add(&dbus_ctx,
+			service_name, NULL, BLUEALSA_INTERFACE_MANAGER, "PCMRemoved", NULL);
+
+	if (!dbus_connection_add_filter(dbus_ctx.conn, dbus_signal_handler, NULL, NULL)) {
+		print_error("Couldn't add D-Bus filter");
+		return EXIT_FAILURE;
+	}
+
+	while (dbus_connection_read_write_dispatch(dbus_ctx.conn, -1));
+
+	return EXIT_SUCCESS;
+}
+
+static struct command {
+	const char *name;
+	int (*func)(int argc, char *arg[]);
+	const char *args;
+	const char *help;
+} commands[] = {
+	{ "list-pcms", cmd_list_pcms, "", "List all PCM paths" },
+	{ "info", cmd_info, "<pcm-path>", "Show PCM properties etc" },
+	{ "codec", cmd_codec, "<pcm-path> [<codec>]", "Change codec used by PCM" },
+	{ "volume", cmd_volume, "<pcm-path> [<val>] [<val>]", "Set audio volume" },
+	{ "mute", cmd_mute, "<pcm-path> [y|n] [y|n]", "Mute/unmute audio" },
+	{ "soft-volume", cmd_softvol, "<pcm-path> [y|n]", "Enable/disable SoftVolume property" },
+	{ "monitor", cmd_monitor, "", "Display PCMAdded and PCMRemoved signals" },
+	{ "open", cmd_open, "<pcm-path>", "Transfer raw PCM via stdin or stdout" },
+};
+
+static void usage(void) {
+	printf("%s - Utility to issue BlueALSA API commands\n", progname);
+	printf("\nUsage:\n  %s [options] <command> [command-args]\n", progname);
+	printf("\nOptions:\n");
+	printf("  -h, --help          Show this help\n");
+	printf("  -V, --version       Show version\n");
+	printf("  -B, --dbus=NAME     BlueALSA service name suffix\n");
+	printf("  -q, --quiet         Do not print any error messages\n");
+	printf("\nCommands:\n");
+	int index;
+	for (index = 0; index < ARRAYSIZE(commands); index++) {
+		printf("  %-12s%-27s%s\n", commands[index].name, commands[index].args, commands[index].help);
+	}
+
+}
+
+int main(int argc, char *argv[]) {
+
+	const struct option long_options[] = {
+		{"dbus",        required_argument, NULL, 'B'},
+		{"version",     no_argument, NULL, 'V'},
+		{"help",        no_argument, NULL, 'h'},
+		{"quiet",       no_argument, NULL, 'q'},
+		{ 0 },
+	};
+
+	progname = argv[0];
+
+	log_open(argv[0], false, false);
+
+	int c;
+	while ((c = getopt_long(argc, argv, "+B:Vhq", long_options, NULL)) != -1) {
+		switch (c) {
+		case 'h' :
+			usage();
+			return EXIT_SUCCESS;
+		case 'V' :
+			printf("%s %s\n", progname, PACKAGE_VERSION);
+			return EXIT_SUCCESS;
+		case 'B' :
+			snprintf(service_name, sizeof(service_name), BLUEALSA_SERVICE ".%s", optarg);
+			break;
+		case 'q' :
+			quiet = true;
+			break;
+		}
```

Comment:
Add `default` case with help info for consistency with other programs in the project:
```c
		default:
			fprintf(stderr, "Try '%s --help' for more information.\n", argv[0]);
			return EXIT_FAILURE;
```